rules/no-for-loop.js

```diff
@@ -220,19 +220,22 @@ const nodeContains = (ancestor, descendant) => {
 	return false;
 };
 
-const isIndexIdentifierUsedElsewhereInTheLoopBody = (forStatement, indexIdentifierName, bodyScope) => {
-	const indexVariable = resolveIdentifierName(indexIdentifierName, bodyScope);
+const isIndexVariableUsedElsewhereInTheLoopBody = (indexVariable, bodyScope) => {
 	const inBodyReferences = indexVariable.references.filter(reference => scopeContains(bodyScope, reference.from));
 
 	// One reference in the body would be the one in the element variable declaration like `const el = arr[i]`.
 	// Any reference besides that should retain the index variable.
 	return inBodyReferences.length > 1;
 };
 
-const someIdentifiersLeakOutOfTheLoop = (forStatement, identifiers, forScope, bodyScope) => {
-	return identifiers.some(name => {
-		const variable = resolveIdentifierName(name, bodyScope);
+const isIndexVariableAssignedToInTheLoopBody = (indexVariable, bodyScope) => {
+	return indexVariable.references
+		.filter(reference => scopeContains(bodyScope, reference.from))
```

Comment:
Not sure if worth combining the two conditions?
```js
return indexVariable.references
	.some(reference => scopeContains(bodyScope, reference.from) && reference.isWrite());
```

Reply:
I'm afraid you'll have to make an argument for it. Performance difference is negligible, readability is fine both ways.